### PR TITLE
DNS test scenarios

### PR DESCRIPTION
**Pull request by sasurobert**
a set of DNS test scenarios - all works as expected

### CODE REVIEW

File: cmd/node/config/genesisSmartContracts.json [hunk 1]
```diff
@@ -11,8 +11,8 @@
     "owner": "erd16grmckn46ry7fwyvass8e8pz88klazzpc0c5f0pnrv643td4797sgnvjkm",
     "filename": "./config/genesisContracts/dns.wasm",
     "vm-type": "0500",
-    "init-parameters": "056bc75e2d63100000",
+    "init-parameters": "00",
```
Comment (camilbancioiu) [inline cmd/node/config/genesisSmartContracts.json:14]: What has changed here?

File: process/smartContract/builtInFunctions/saveUserName.go [hunk 2]
```diff
@@ -60,7 +58,7 @@ func (s *saveUserName) ProcessBuiltinFunction(
 	if !ok {
 		return nil, process.ErrCallerIsNotTheDNSAddress
 	}
-	if len(vmInput.Arguments) != 1 || len(vmInput.Arguments[0]) != userNameHashLength {
+	if len(vmInput.Arguments) != 1 {
```
Comment (camilbancioiu) [inline process/smartContract/builtInFunctions/saveUserName.go:61]: A maximum length would be good here.

**Review by bogdan-rosianu — APPROVED** [review]

File: process/smartContract/builtInFunctions/saveUserName.go (hunk 2 above)
Comment (sasurobert) [reply process/smartContract/builtInFunctions/saveUserName.go:61]: No need. We control the DNS SCs which are only allowed to call this

**Review by camilbancioiu — APPROVED** [review]

**Review by LucianMincu — APPROVED** [review]
System tests passed.

**Pull request merged**